src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -60,10 +61,41 @@ public ISonarQubeServer CreateSonarQubeServer(ProcessedArgs args)
             var password = args.GetSetting(SonarProperties.SonarPassword, null);
             var hostUrl = args.SonarQubeUrl;
 
-            this.server = new SonarWebService(new WebClientDownloader(username, password, this.logger), hostUrl, this.logger);
+            var proxyHost = ProxyParser(args.GetSetting(SonarProperties.Proxy, null));
+            WebProxy wp = null;
+            if ( proxyHost.Length==2 )
+            {
+                Console.WriteLine("Proxy setup Starting ...");
+                wp = new WebProxy(proxyHost[0], int.Parse(proxyHost[1]));
+                Console.WriteLine("Setting host");
```

Comment:
The `logger` should be used here instead.

Reply:
Done